packages/back-end/src/events/base-events.ts

```diff
@@ -0,0 +1,26 @@
+import { NotificationEventPayload } from "./base-types";
+import { ApiFeatureInterface } from "../../types/api";
+
+/**
+ * Some resource IDs are not unique. This can help.
+ * Example:   ApiFeatureInterface & WithOrganizationId
+ */
+export type WithOrganizationId = { organizationId: string };
```

Comment:
Instead of adding organizationId to `data`, should we make that a top-level field in the event payload instead?  We will likely want to add a Mongo index on org id, which is easier to do as a top-level field that's always present.

Reply:
I would be reluctant to do so in case we want events to be relevant for events that are not related to organizations and may be related to the user specifically. For example, if we wanted to start tracking account-related events, e.g. login events from different devices, password change, profile updates, etc. (like the security features many other services offer), I don't think it would make sense to couple events to organizations. In that case, the relevant data on the event could be the user's ID.

Reply:
Also I just realized I'm already tightly coupling to the organization. I think it was something I needed to add because feature ID's are not unique. I can add an index and we can worry about how we track events not related to organizations at another time.